core/consensus/common/src/import_queue.rs

```diff
@@ -229,14 +256,20 @@ struct BlockImporter<B: BlockT> {
 	result_port: Receiver<BlockImportWorkerMsg<B>>,
 	worker_sender: Sender<BlockImportWorkerMsg<B>>,
 	link: Option<Box<dyn Link<B>>>,
+	verifier: Arc<Verifier<B>>,
```

Comment:
Maybe finality proof verification should be moved to the `BlockImportWorker` since it may be a bit heavy (and might require importing extra headers).

Reply:
Good point, thanks! Will (try to) do